client/ayon_flame/api/plugin.py

```diff
@@ -631,46 +637,156 @@ def get_native_colorspace(cls, input_colorspace):
+                "Unable to import any other versions alongside the current "
+                "selection.\nThis may result in unordered versions within the"
+                " clip. Deleting the clip and reloading it will resolve this "
+                "issue.",
+                "warning",
+                ["OK"],
+            )
+            all_versions = [version_entity]
+        else:
+            all_versions = ayon_api.get_versions(
+                project_name,
+                product_ids=[version_entity["productId"]],
+                task_ids=[version_entity["taskId"]],
+            )
 
-        # new feed variables:
-        feed_path = feed_data.pop("path")
+        for version in all_versions:
+            representation = ayon_api.get_representation_by_name(
+                project_name,
+                representation_name,
+                version_id=version["id"],
+            )
+
+            version_context = deepcopy(context)
+            version_context["version"] = version
+            version_context["representation"] = representation
+
+            version_name = f"v{version['version']:03}"
+            colorspace = self.get_colorspace(version_context)
 
-        # initialize parent class
-        super(OpenClipSolver, self).__init__(
-            feed_path,
-            logger=logger
+            layer_rename_template = self.layer_rename_template
+
+            # in case output is not in context replace key to representation
+            if not representation["context"].get("output"):
+                layer_rename_template = self.layer_rename_template.replace(
+                    "output", "representation"
+                )
+
+            # convert colorspace with ocio to flame mapping
+            # in imageio flame section
+            colorspace = self.get_native_colorspace(colorspace)
+
+            # prepare clip data from context ad send it to openClipLoader
+            path = self.filepath_from_context(version_context)
+
+            clip_solver.add_feed(
+                path,
+                version_name,
+                colorspace,
+                representation["context"],
+                layer_rename_template,
+            )
+
+        clip_solver.set_current_version(
+            f"v{version_entity['version']:03}"
+        )
+
+        clip_solver.write()
+
+        # prepare Reel group in actual desktop
+        opc = self._get_clip(
+            clip_name,
+            openclip_path
         )
 
-        # get other metadata
-        self.feed_version_name = feed_data["version"]
-        self.feed_colorspace = feed_data.get("colorspace")
-        self.log.debug("feed_version_name: {}".format(self.feed_version_name))
+        if len(all_versions) == 1 and rename_clip:
+            opc.name = f"{clip_name} [v{version_entity['version']:03}]"
+        else:
+            opc.name = clip_name
+
+        if message_args:
+            flame.messages.show_in_dialog(*message_args)
```

Comment:
```suggestion
        if not version_entity["taskId"]:
            flame.messages.show_in_dialog(
                "Import Warning",
                (
                    "Version doesn't have set task."
                    "\nUnable to import any other versions alongside"
                    " the current selection."
                    "\nThis may result in unordered versions within the"
                    " clip. Deleting the clip and reloading it will resolve"
                    " this issue."
                ),
                "warning",
                ["OK"],
            )
```